### ISSUE TITLE
ruff ignores undefined function

### ISSUE THREAD

**Issue by dstromberg** [astral-sh/ruff-lsp#488]
Hi.

This tiny script reproduces the issue for me both in astronvim with ruff-lsp and with "ruff check":
```
#!/usr/local/cpython-3.10/bin/python3

"""Test script."""

fred()
```

That is, the undefined function "fred" is not flagged as an error.

My ruff version is 0.0.261

My ~/.config/ruff/pyproject.toml has:
```
[tool.ruff]
line-length = 132

# Ignore global imports that aren't at the top of a file
lint.ignore = ["E402"]

# "D": Enable all `pydocstyle` rules, limiting to those that adhere to the PEP-257 convention via `convention = "pep257"`, below.
# "ARG001" Enable checking unused function/method arguments.
# "C901" Enable checking McCabe Complexity.
lint.select = ["D", "ARG001", "C901"]

[tool.ruff.lint.pydocstyle]
convention = "pep257"

[tool.ruff.lint.per-file-ignores]
# Ignore set-but-not-used in __init__.py files
"__init__.py" = ["F401"]

[tool.ruff.lint.mccabe]
# I believe this defaults to 10.  With a setting of 10 like this, it should warn if the complexity is >= 11.
max-complexity = 10

```

I'm almost sure this was working previously.

Any suggestions?

Thanks for reading, and hopefully for responding too!

**Comment by dstromberg** [comment]
BTW, I believe ruff-lsp is working at least partially, because if I insert !! it gets flagged, and if I add a function with an unused argument that's flagged too.

**Comment by dhruvmanila** [comment]
> My ruff version is 0.0.261

That seems like a pretty old Ruff version. Can you try upgrading to the latest version (`0.6.4`) and check if the issue is present?

**Comment by dstromberg** [comment]
My mistake: my mason version of ruff and ruff-lsp are:
$ python3 -m pip list -v | grep '^ruff'
below cmd output started 2024 Fri Sep 06 07:12:19 AM PDT
ruff              0.6.4    /data/home/dstromberg/.local/share/nvim/mason/packages/ruff-lsp/venv/lib/python3.10/site-packages pip
ruff-lsp          0.0.55   /data/home/dstromberg/.local/share/nvim/mason/packages/ruff-lsp/venv/lib/python3.10/site-packages pip

**Comment by dstromberg** [comment]
I removed /usr/local/bin/ruff too, even though I think it was the mason version of ruff that mattered.

**Comment by MichaReiser** [comment]
I think the problem here is that you don't enable the `F821` rule which catches undefined names. Maybe you wanted to use `extend-select` instead of `select`? `select` overrides the defaults whereas `extend-select` selects additional rules to the default once.

**Comment by dstromberg** [comment]
That was the problem.  Thanks!
